bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTreeViewer.java

```diff
@@ -140,6 +155,40 @@ protected AbstractTreeViewer() {
 		// do nothing
 	}
 
+	/**
+	 * Expands a widget and creates it's children.
+	 *
+	 * @param w Widget to expand
+	 * @return The children of the Widget
+	 * @since 3.31
+	 */
+	protected Item[] expandAndCreateChildren(Widget w) {
+		setExpanded((Item) w, true);
+		createChildren(w);
+		return getChildren(w);
```

Comment:
This seems to reimplement the essential expansion functionality as already implemented in `setExpandedState`. Is there a necessity for this additional method?